sandbox/sandbox/src/lib.rs

```diff
@@ -89,14 +91,36 @@ impl From<Error> for HostError {
 /// supervisor in [`EnvironmentDefinitionBuilder`].
 ///
 /// [`EnvironmentDefinitionBuilder`]: struct.EnvironmentDefinitionBuilder.html
-pub type HostFuncType<T> = fn(&mut T, &[Value]) -> Result<env::WasmReturnValue, HostError>;
+pub type HostFuncType<T> =
+    fn(&mut default_executor::Caller<'_, T>, &[Value]) -> Result<env::WasmReturnValue, HostError>;
+
+/// Sandbox store.
+pub trait SandboxStore<T>: AsContext<T> {
+    /// Create a new sandbox store.
+    fn new(state: T) -> Self;
+}
+
+/// Sandbox caller.
+pub trait SandboxCaller<T>: AsContext<T> {
+    /// Set WASM global.
+    fn set_global_val(&mut self, name: &str, value: Value) -> Option<()>;
+
+    /// Get WASM global.
+    fn get_global_val(&self, name: &str) -> Option<Value>;
+}
```

Comment:
trait's methods aren't used anywhere so I guess we can remove them or even the trait altogether